flake.nix

```diff
@@ -0,0 +1,53 @@
+{
+  description = "Flake for Plutus";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs?rev=7d71001b796340b219d1bfa8552c81995017544a";
+    haskell-nix.url = "github:input-output-hk/haskell.nix?rev=87084d65a476cc826a0e8c5d281d494254f5bc7a";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, nixpkgs, haskell-nix, flake-utils, ... }:
+    (flake-utils.lib.eachSystem [ "x86_64-linux" ] (system:
+      let
+        #
+        # Obtain all niv sources extending it with the 'nixpkgs' flake input
+        #
+        sources =
+          let
+            sourcesInfo = builtins.fromJSON (builtins.readFile ./nix/sources.json);
+            fetch = sourceInfo: builtins.fetchTarball { inherit (sourceInfo) url sha256; };
+          in
+          builtins.mapAttrs (name: info: fetch info) sourcesInfo // { inherit nixpkgs; };
+
+        #
+        # all packages from nix/default.nix
+        #
+        plutusPackages = import ./nix {
+          inherit system sources;
+          haskellNixOverlays = [ haskell-nix.overlay ];
+        };
+
+        #
+        # all packages from ./default.nix
+        #
+        topLevel = import ./. {
+          inherit system;
+          packages = plutusPackages;
+        };
+
+        inherit (plutusPackages) pkgs plutus ownOverlays;
+        inherit (plutus) haskell iohkNix;
+        inherit (plutus.lib) buildPursPackage buildNodeModules filterNpm gitignore-nix;
+      in
+      rec {
+        packages = rec {
```

Comment:
Should we be exporting the nixos system configurations too?

Reply:
No, unless i am mistaken those are for the bin in the context of `bitte` i'm afraid.

What we actually need to provide though are shell scripts per relevant service as we won't have nixos services anymore either. I will have to take care of those next.

Reply:
No nixos services!